ambry-api/src/main/java/com.github.ambry/account/Container.java

```diff
@@ -324,28 +442,37 @@ public boolean equals(Object o) {
     if (this == o) {
       return true;
     }
-    if (!(o instanceof Container)) {
+    if (o == null || getClass() != o.getClass()) {
```

Comment:
Yes, seems like the "intellij default generator does it this way. This ensures that the class matches exactly instead of possibly having a class/subclass relationship.